seq: Performance optimization

FidelSch opened the pull request:
Reading #6182 I noticed that most of the time spent running `cargo run seq  4e4000003 4e4000003` was just BigUint::to_string()

Reading the code I found it is being called twice, once to get the actual string representation of the first number, and again on the last number, but only to get its length; and discarding the actual result.
This seems fine on fairly small numbers, but its efficiency degrades significantly on larger ones.

In my machine, this change resulted in a ~2x speedup on the mentioned case, and a marginal but seemingly better performance on smaller cases.

```bash
$ ~/coreutils$ hyperfine -L seq target/release/coreutils,target/release/coreutils_old "{seq} seq 4e4000003 4e4000003" 
Benchmark 1: target/release/coreutils seq 4e4000003 4e4000003
  Time (mean ± σ):     26.009 s ±  0.113 s    [User: 25.992 s, System: 0.015 s]
  Range (min … max):   25.909 s … 26.294 s    10 runs
 
Benchmark 2: target/release/coreutils_old seq 4e4000003 4e4000003
  Time (mean ± σ):     52.372 s ±  0.446 s    [User: 52.352 s, System: 0.017 s]
  Range (min … max):   51.815 s … 53.017 s    10 runs
 
Summary
  'target/release/coreutils seq 4e4000003 4e4000003' ran
    2.01 ± 0.02 times faster than 'target/release/coreutils_old seq 4e4000003 4e4000003'
```

FidelSch committed:
seq: calculate number length more efficiently

ChrisDryden commented:
Would be great to add your example to the benchmarks

anastygnome reviewed (changes requested):
Any reason for not using

```rust 
n.checked_ilog10().unwrap_or(0) + 1
```

Which should be available?

sylvestre commented:
any idea why tsort_input_parsing_heavy regressed ?

FidelSch committed:
Merge branch 'uutils:main' into seq-optimization

anastygnome commented:
> > Any reason for not using
> > ```rust
> > n.checked_ilog10().unwrap_or(0) + 1
> > ```
> 
> Seemed unnecessary given it is just a constant. If there is any benefit to this alternative I am happy to refactor.

If I recall correctly the performance is similar to your solution and it's part of the language. Could you try sending a commit to trigger the benchmarks with this solution instead?

sylvestre commented:
> After some digging, the maximum value `seq` accepts seems to be `11e4931`, equivalent to the maximum value representable by an 80-bit `long double`; which supports the theory that GNU uses this type for their implementation. By using BigDecimal we are significantly extending the representable range, so it makes sense that for huge numbers a direct comparison to `seq` is not viable.

ok, could you please document this in docs/src/extensions.md ? thanks

sylvestre committed:
Merge branch 'main' into seq-optimization

FidelSch committed:
seq: Update extensions.md 

Clarify seq output accuracy and value range limitations compared to GNU coreutils.

sylvestre reviewed (commented):
The optimization idea is sound — avoiding `to_string()` on large numbers is a real win.

However, the `bits() / LOG2_10` approximation can be off by 1 for some values (e.g., exact powers of 10). Since this is used for padding width, being off by one character could produce misaligned output. Have you verified against the GNU test suite?
